sds_data_manager/lambda_code/SDSCode/pipeline_lambdas/spice_indexer.py

```diff
@@ -0,0 +1,386 @@
+            if spice_metadata["end_date"] is None:
+                spice_metadata["end_date"] = maximum_mission_time
+            file_coverage_datetime = [
+                [spice_metadata["start_date"], spice_metadata["end_date"]]
+            ]
+            file_coverage_j2000 = [
+                [
+                    spiceypy.datetime2et(spice_metadata["start_date"]),
+                    spiceypy.datetime2et(spice_metadata["end_date"]),
+                ]
+            ]
+            file_coverage_sclk = [
+                [
+                    spiceypy.sce2s(SPACECRAFT_ID, file_coverage_j2000[0][0]),
+                    spiceypy.sce2s(SPACECRAFT_ID, file_coverage_j2000[0][1]),
+                ]
+            ]
+        else:
+            function_arguments = {
+                "idcode": SPACECRAFT_ID * 1000,
+                "cover": spiceypy.cell_double(COVERAGE_SPICE_ARRAY_LENGTH),
+            }
+            if "attitude" in spice_metadata["type"]:  # Extra arguments needed for ckcov
+                function_arguments["needav"] = COVERAGE_ANGULAR_VELOCITY_ONLY
+                function_arguments["level"] = COVERAGE_LEVEL
+                function_arguments["tol"] = COVERAGE_TOLERANCE
+                function_arguments["timsys"] = COVERAGE_TIME_SYSTEM
+            file_coverage_j2000, file_coverage_datetime, file_coverage_sclk = (
+                get_coverage_dictionary(spice_file, **function_arguments)
+            )
+
+    # Insert/Update the gathered data into the database
+    _upsert_into_spice_table(
+        spice_object,
+        file_coverage_j2000,
+        file_coverage_datetime,
+        file_coverage_sclk,
+        latest_lsk,
+        latest_sclk,
+    )
+
+
+def create_symlink(source_path: Path, destination_path: Path) -> None:
+    """Create a symlink from source_path to destination_path.
+
+    Parameters
+    ----------
+    source_path : str
+        Source path of the symlink
+    destination_path : str
+        Destination path of the symlink
+
+    """
+    # Remove the old symlink
+    destination_path.unlink(missing_ok=True)
+
+    # Create a new symlink pointing to the new file
+    destination_path.symlink_to(source_path)
+
+
+def write_data_to_efs(s3_key: str, s3_bucket: str, spice_mount_path: Path) -> Path:
+    """Write data to EFS and create/update symlink.
+
+    Parameters
+    ----------
+    s3_key : str
+        S3 object key
+    s3_bucket : str
+        The S3 bucket
+    spice_mount_path: Path
+        The path to the local SPICE directory
+
+    Returns
+    -------
+    efs_spice_filename_and_path : Path
+        The local location of the SPICE file
+
+    """
+    # Create an S3 client
+    s3_client = boto3.client("s3")
+
+    # Remove 'spice/' prefix from the s3 key. See key example below.
+    #   Eg. spice/spin/imap_2025_122_2025_122_02.spin.csv
+    # Keep remaining folder path after `spice/` to match the folder structure
+    # defined in imap-data-access library.
```

Comment:
I'm not sure I follow this portion. I thought our file structure was supposed to be the same in s3 as on disk?

Can you set `imap_data_access.config["DATA_DIR"] = spice_mount_path.parent` and then create the `imap_data_access.SPICEFilePath(os.path.basename(s3_key))` which I think would fill out the proper directory structures for you to be consistent everywhere?